conda_build/build.py

```diff
@@ -1276,7 +1276,8 @@ def test(recipedir_or_package_or_metadata, config, move_broken=True):
 
     warn_on_use_of_SRC_DIR(metadata)
 
-    config.compute_build_id(metadata.name())
+    if config.set_build_id:
```

Comment:
I think that without --dirty, the work dir would be cleaned up, but to be honest, I really never use --no-build-id.  --no-build-id definitely breaks multiple builds happening at once, but it should still be clean for one build at a time.